### ISSUE TITLE
[Feedback][Feature Request] SSL/TLS Support for CasaOS.local

### ISSUE THREAD

**Issue by strickdd** [IceWhaleTech/CasaOS#1074]
### Description

Forgive me if my Google-fu isn't up to par, but I haven't been able to find a way to set up SSL/TLS for http://casaos.local > https://casaos.local. 

It would be nice to have, at a minimum, a self-signed cert that gets created per install and have that used for the UI access. Taking a step further, allowing for use of ACME (Let's Encrypt) to use a customer domain with a trusted CA-signed cert.

### Additional Information

	 - OS: linux
	 - CasaOS Version: 0.4.3
	 - Disk Total: 27712 
	 - Disk Used: 16408 
	 - System Info: [Debian GNU/Linux 11   \l Bit:64 Mem:7801M Core:4 Linux CasaOS 5.10.0-10-amd64 #1 SMP Debian 5.10.84-1 (2021-12-08) x86_64 GNU/Linux]
	 - Remote Version: 0.4.3
	 - Browser: Chrome 
	 - Version: 112.0.0.0

**Comment by tigerinus** [comment]
Looking into implementing this via integration with Traefik

related: #773 #329

**Comment by nghialele** [comment]
Any updates on this?

**Comment by tigerinus** [comment]
> Any updates on this?

We discussed about this internally, and concluded that it's not the best time to integrate with Traefik at the moment.

That said, this feature request is still valid, and will stay open and pending to be prioritized by @ETWang1991 who manages and coordinate all feature requests in our roadmap.

**Comment by tigerinus** [comment]
A potential pkg we could use for this is https://github.com/go-acme/lego

**Comment by mdanlowski** [comment]
➕ 1️⃣

**Comment by lucaberti** [comment]
Just like to add my support for this feature request! Thank you for your work!

**Comment by Plarpoon** [comment]
Yeah, this would be nice, in the meantime I will find a manual way to do it on my own environment but having this feature as default would be really good. Especially for newer users to the homelab world.

**Comment by k3z** [comment]
Same here. Little frustation after installing unify controller.
Great work with casaOS (plus in my case a zimaboard).
Thanks for all this awesomeness.

**Comment by edremington** [comment]
+1

**Comment by dwt** [comment]
[It is possible to just run lets encrypt on the base system and integrate it with the running Webserver (apache by default)](https://wiki.casaos.io/en/guides/casaos-behind-webserver#h-51-external-ca) Does that help?

**Comment by hostmaxonline** [comment]
+1

**Comment by mavenboard** [comment]
any updates?

**Comment by jinhucheung** [comment]
+1

**Comment by DznDani** [comment]
+1

**Comment by marchingon12** [comment]
+1

**Comment by sg0luc** [comment]
+1

**Comment by dinhdobathi1992** [comment]
+1

**Comment by LightYagami28** [comment]
+1?

**Comment by apiaskowski** [comment]
Hi, has anyone got Tailscale working and exposing the CasaOS WebUI, e.g. by "serve"  function or their proxy?
But yeah, manually providing an own SSL/TLS certificate would be best option.

**Comment by rjroot** [comment]
+1

**Comment by miiraheart** [comment]
+1

**Comment by pcarbonel** [comment]
+1

**Comment by LightYagami28** [comment]
I switch to UmbrelOs NOW :)

**Comment by Manchild25** [comment]
+1, would like to use NPM with my domain to access the UI over TLS.

**Comment by inkly** [comment]
Partially, and deliberately so: the inkly distribution of CasaOS (v0.4.40) lets the gateway serve HTTPS with a certificate you supply — `tlscert` and `tlskey` in `/etc/casaos/gateway.ini`. That covers what most of this thread asks for; several people here already run a reverse proxy or certbot and just want to hand the result to CasaOS.

There is no ACME in the gateway. HTTP-01 needs port 80 reachable from the internet, which a LAN box rarely is, and that job belongs in Caddy or Traefik in front.

```
curl -fsSL https://github.com/inkly/CasaOS-Install/releases/latest/download/install.sh | sudo bash
```

Details: https://github.com/inkly/CasaOS-Install#what-is-in-v0440
